lib/autocomplete.js

```diff
@@ -0,0 +1,90 @@
+const _ = require("lodash");
+const aws = require("aws-sdk");
+const parsers = require("./parsers");
+const core = require("./core");
+const consts = require("./consts.json");
+
+/**
+ * Maps parameters passed to Autocomplete function to an object with parsed values.
+ *
+ * @param params: [ { value: any, name: string, type: string } ]
+ * Array of values passed to autocomplete function as action parameters.
+ *
+ * @returns {*} - object with params names as keys and parsed values
+ */
+function parseActionParameters(params) {
+  return params.reduce((acc, {
+    value, name, type, valueType,
+  }) => ({
+    ...acc,
+    [name]: parsers.resolveParser(type || valueType)(value),
+  }), {});
+}
+
+function handleInput(awsService, actionParams, pluginSettings) {
+  const [params, settings] = [actionParams, pluginSettings].map(parseActionParameters);
+  const client = core.getServiceInstance(awsService, params, settings);
+  return { params, settings, client };
+}
+
+function simpleItemsFilter(autocompleteItems, query) {
```

Comment:
I've concluded that this function filters `simpleItems`. Instead of saying what it performs, explain what it achieves, what the result is (eg. alphabetically ordered list of query words, in case of the `else` block).

Is `simpleItem` some specific universal structure?

Reply:
The intention was rather it's a simple "itemsFilter" rather than "simpleItems" filter, but I get why it's confusing. I'll think about how to make it more meaningful.